### DISCUSSION TITLE
Additional shortcut keys (and ideas) to help with navigation

### DISCUSSION THREAD

**Discussion by NyarlathotepZERO** [FakeFishGames/Barotrauma#4691] [Ideas and suggestions]
I propose that the availability of some shortcut keys be provided, to assist with piloting the sub through the navigation terminal.

1. A shortcut key for manual ping of sonar, or at least a key that can change the toggle state of the sonar switch.
**Reason:** This will save having to drag the mouse across the screen to double click the sonar every 10 seconds or so, just to make it pulse on/off. The toggle switch is there for those captains who want to light up their sub like a christmas tree, attracting the attention of everything within 100+ meters. But for those captains who are more inclined towards a stealthy approach, it provides a way of keeping the mouse cursor over more important things. To add to this, the ability to control the directional ping's 360 degree radius with something more tactile like the mousewheel could be a really pleasant way of implementing the directional ping feature in a way that is a little less 'clicky'.

2. A shortcut key to bring the submarine to a halt in the same way that flicking the toggle from manual control to autopilot does.
**Reason:** When manually piloting, it would be beneficial to be able to hit a key to recenter steering bringing the sub to a halt over its current position. The number of reasons for doing this are many as I'm sure you can guess, but to throw a few out there, a few may include; positioning the sub over a specific spot on a wreck, quickly jumping from a navigating char to a char of another profession or even just pulling up the sub hard due to the sudden appearance of an underwater structure, desired resource, nasty creature within perfect turret radius, shout from a fellow crewmember regarding the still missing diving suit as you leave the wreck, etc.

3. Shortcut keys to jump straight to NPC crewmen.
**Reason:** Oftentimes you need to jump from your current char to another NPC. Currently as it stands our crew are ordered in the way that they are purchased without any way of changing this order (which by the way, is also something I'd also love to see changed). It would benefit players greatly to have the option of being able to jump instantly from navigating, straight to a particular char without having to run down through the entire list to finally realize you jumped one or 2 over... and then need to jump back a couple. To add to this, it could be nice to see on the status monitor screen, where the crewmembers actually are. perhaps this could be some sort of 'status report' showing an icon that only updates upon request.

A couple of other ideas that I had in regards to managing NPC would be giving us the option to remove or clear entirely the job history shortcut buttons in the char navbar. Reason being, its frustrating to clear up problems when a char has been assigned to the wrong job (especially a wrong weapon) This could also probably be helped slightly by giving each gun a unique colour or number.

Also it'd be very stress relieving to have this job history assignment save on each char between docking/missions. This would mean we wouldn't have to reassign the exact same thing to each and every char EVERY time we jumped back in the sub for another dive.

Most people these days have mice/keyboards with a plethora of macro buttons just waiting to be assigned so that we can tailor our unique experience of games, provided we have the option in games to do so.

**Comment by Saevon** [comment]
1. Is great, right clicking to send a directional ping would be awesome. But keep it in passive otherwise. Right clicking on the sub itself (any compartment) sends a non-directional pulse?

2. is actually doable using circuits. Put a switch right outside nav that sets x/y to 0 while pressed!  Then you cancel nav and quickly hit the switch?

**Comment by Saevon** [comment]
see #4466  for a way to toggle sonar passive/active. Making it easier to send a pulse out

**Comment by someone972** [comment]
Another +1 for having at minimum a key to toggle between active/passive, if not a dedicated key to send out a single pulse. Our group has been wanting a "single pulse" behavior for a while now. And anything that makes directional ping easier to use is welcome, as at our skill level with navigating we pretty much never use the directional.

**Comment by Benzin73** [comment]
Yet another +1.
Fully agree on all points! it would be a **huge** QoL improvement! specially for Single player users like myself!

**Comment by 3e849f2e5c** [comment]
Completely agree on the active/passive sonar switch keybind after playing thru the entire campaign as a captain

**Comment by James-E-A** [comment]
Yes, I would actually use the directional sonar if they didn't have such a cursed control scheme

1. Mouse over and toggle directional on
2. Mouse over and set direction
3. Mouse over and ping active sonar
4. Repeat steps 2–3 until done
5. Mouse over and toggle directional off

This is simply too torturous to use if there's any reasonable alternative—I would love to see more "IRL" controls (i.e. keyboard or mouse-buttons) added to the entire nav panel, including directional sonar, autopilot-toggle, and autopilot-position

**Comment by marpul358** [comment]
Second one already exists as the "autopilot" switch flick. But is has a problem when using some type of ships.
For example the Winterhalter and you have full throttle forward. You click it on autopilot to "stop" the ship. Afterwards you flick the autopilot off and the manual throttle is still full throttle forward which causes the junction boxes to melt.

Would be a nice QoL to have two new separate buttons - one to set throttle to idle irrespective of the status of the manual/autopilot switch and the other to set the buoyancy/depth to zero/neutral and that as well irrespective of the autopilot switch.

**Reply by James-E-A** [reply]
See #7089 for this particular aspect
